e2e/files.spec.ts

```diff
@@ -0,0 +1,283 @@
+      }
+
+      const userName = userConfigs['swc-e2e-user'].username
+      const validatedUserName = userConfigs['swc-e2e-user-validated'].username
+
+      const { fileEntityId, fileName } = await testAuth.step(
+        'Create a file',
+        async () => {
+          const userAccessToken = await getAccessTokenFromCookie(userPage)
+          const fileName = `${generateEntityName('file')}.txt`
+          const { fileEntityId, fileHandleId } = await createFile(
+            fileName,
+            'Created for SWC e2e testing.',
+            'text/txt',
+            userProject.id,
+            userAccessToken,
+            userPage,
+          )
+          fileHandleIds.push(fileHandleId)
+          return { fileEntityId, fileName }
+        },
+      )
+
+      await testAuth.step('First user can view a private file', async () => {
+        await userPage.goto(`${entityUrlPathname(userProject.id)}/files`)
+        await waitForInitialPageLoad(userPage)
+
+        const fileLink = userPage.getByRole('link', { name: fileName })
+        await expect(fileLink).toBeVisible()
+
+        await fileLink.click()
+        await expect(
+          userPage.getByText(`Discussion about ${fileName}`),
+        ).toBeVisible()
+      })
+
+      await openFileSharingSettings(userPage)
+      await confirmSharingSettings(userPage, userName, validatedUserName, false)
+
+      await testAuth.step('Second user cannot access the file', async () => {
+        await validatedUserPage.goto(entityUrlPathname(fileEntityId))
+        await waitForInitialPageLoad(validatedUserPage)
+
+        await expectNoAccessPage(validatedUserPage)
+      })
+
+      await testAuth.step(
+        'First user enables Local Sharing Settings for the File',
+        async () => {
+          await userPage
+            .getByRole('button', { name: 'Create Local Sharing Settings' })
+            .click()
+          await expect(
+            userPage.getByRole('heading', { name: 'Add More People' }),
+          ).toBeVisible()
+        },
+      )
+
+      await testAuth.step(
+        'First user shares the file with second user',
+        async () => {
+          const userSuggestBox = userPage.getByRole('textbox', {
+            name: 'Enter name...',
+          })
+          await userSuggestBox.fill(validatedUserName)
+          await userSuggestBox.press('Shift')
+
+          await userPage
+            .getByRole('menuitem', { name: `@${validatedUserName}` })
+            .click({ timeout: testInfo.timeout * 3 })
+
+          await expect(
+            userPage.getByRole('cell', { name: validatedUserName }),
+          ).toBeVisible({ timeout: testInfo.timeout * 3 })
+
+          // Don't send message, so we don't have to clean up the message
+          await userPage.getByText('Notify people via email').click()
+
+          // Cookies banner covers the button to save settings
+          // ...so close banner by accepting cookies
+          await acceptSiteCookies(userPage)
```

Comment:
Would it make sense to move this into a fixture, and just auto-accept for every new page (user?) instance? I don't think we should have to think about this in this test.

Reply:
Sure, I'll add it in to the user page fixtures!